deps/rabbit/src/rabbit_stream_sac_coordinator.erl

```diff
@@ -0,0 +1,756 @@
+    rabbit_log:debug("New consumer ~p ~p in group ~p, partition index "
+                     "is ~p",
+                     [ConnectionPid,
+                      SubscriptionId,
+                      {VirtualHost, Stream, ConsumerName},
+                      PartitionIndex]),
+    StreamGroups1 =
+        maybe_create_group(VirtualHost,
+                           Stream,
+                           PartitionIndex,
+                           ConsumerName,
+                           StreamGroups0),
+
+    do_register_consumer(VirtualHost,
+                         Stream,
+                         PartitionIndex,
+                         ConsumerName,
+                         ConnectionPid,
+                         Owner,
+                         SubscriptionId,
+                         State#?MODULE{groups = StreamGroups1});
+apply({unregister_consumer,
+       VirtualHost,
+       Stream,
+       ConsumerName,
+       ConnectionPid,
+       SubscriptionId},
+      #?MODULE{groups = StreamGroups0} = State0) ->
+    {State1, Effects1} =
+        case lookup_group(VirtualHost, Stream, ConsumerName, StreamGroups0) of
+            undefined ->
+                {State0, []};
+            Group0 ->
+                {Group1, Effects} =
+                    case lookup_consumer(ConnectionPid, SubscriptionId, Group0)
+                    of
+                        {value, Consumer} ->
+                            rabbit_log:debug("Unregistering consumer ~p from group",
+                                             [Consumer]),
+                            G1 = remove_from_group(Consumer, Group0),
+                            rabbit_log:debug("Consumer removed from group: ~p",
+                                             [G1]),
+                            handle_consumer_removal(G1, Consumer);
+                        false ->
+                            rabbit_log:debug("Could not find consumer ~p ~p in group ~p ~p ~p",
+                                             [ConnectionPid,
+                                              SubscriptionId,
+                                              VirtualHost,
+                                              Stream,
+                                              ConsumerName]),
+                            {Group0, []}
+                    end,
+                SGS = update_groups(VirtualHost,
+                                    Stream,
+                                    ConsumerName,
+                                    Group1,
+                                    StreamGroups0),
+                {State0#?MODULE{groups = SGS}, Effects}
+        end,
+    {State1, ok, Effects1};
+apply({activate_consumer, VirtualHost, Stream, ConsumerName},
+      #?MODULE{groups = StreamGroups0} = State0) ->
+    {G, Eff} =
+        case lookup_group(VirtualHost, Stream, ConsumerName, StreamGroups0) of
+            undefined ->
+                rabbit_log:warning("trying to activate consumer in group ~p, but "
```

Comment:
I think this will return whatever `rabbit_log:warning` returns (probably `ok`) and crash as it cannot match agains `{G, Eff}`